sway/desktop/output.c

```diff
@@ -573,56 +572,31 @@ static struct output_config *output_config_for_config_head(
 
 static void output_manager_apply(struct sway_server *server,
 		struct wlr_output_configuration_v1 *config, bool test_only) {
-	size_t configs_len = wl_list_length(&root->all_outputs);
-	struct matched_output_config *configs = calloc(configs_len, sizeof(*configs));
+	size_t configs_len = wl_list_length(&config->heads);
+	struct output_config **configs = calloc(configs_len, sizeof(*configs));
 	if (!configs) {
 		return;
 	}
 
 	int config_idx = 0;
-	struct sway_output *sway_output;
-	wl_list_for_each(sway_output, &root->all_outputs, link) {
-		if (sway_output == root->fallback_output) {
-			configs_len--;
-			continue;
-		}
-
-		struct matched_output_config *cfg = &configs[config_idx++];
-		cfg->output = sway_output;
-
-		struct wlr_output_configuration_head_v1 *config_head;
-		wl_list_for_each(config_head, &config->heads, link) {
-			if (config_head->state.output == sway_output->wlr_output) {
-				cfg->config = output_config_for_config_head(config_head, sway_output);
-				break;
-			}
-		}
-		if (!cfg->config) {
-			cfg->config = find_output_config(sway_output);
-		}
+	struct wlr_output_configuration_head_v1 *config_head;
+	wl_list_for_each(config_head, &config->heads, link) {
+		// Generate the configuration and store it as a temporary
+		// config. We keep a record of it so we can remove it later.
+		struct output_config *oc = output_config_for_config_head(config_head);
+		store_temp_output_config(oc);
+		configs[config_idx++] = oc;
 	}
+	configs_len = config_idx;
 
-	sort_output_configs_by_priority(configs, configs_len);
-	bool ok = apply_output_configs(configs, configs_len, test_only, false);
+	bool ok = apply_all_output_configs(test_only, false);
```

Comment:
Maybe we can change this function to accept a pointer to an array of output configs (just pass configs directly) and a length, instead of storing it in output state. Not liking these `store_temp_output_config()`/`remove_temp_output_config()` tbh.

Reply:
That's not a bad idea. It'll take a few signature changes, but should be fine.

Reply:
Take a look at: https://github.com/swaywm/sway/pull/8341